### PR TITLE
cocktail 17.4

### PR DESCRIPTION
**Pull request by khipp**
Created by `brew bump`

---

Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by bevanjkay** [comment]
@SMillerDev I'm not sure why the rosetta audit is failing here, the caveats aren't being read correctly on older versions, or perhaps it is to do with simulating the architecture with `brew audit`?

**Comment by khipp** [comment]
The caveat `requires_rosetta` has a guard clause that checks the contents of the `RUBY_PLATFORM` environment variable:

```ruby
caveat :requires_rosetta do
  next unless Hardware::CPU.arm?

  <<~EOS
    #{@cask} is built for Intel macOS and so requires Rosetta 2 to be installed.
    You can install Rosetta 2 with:
      softwareupdate --install-rosetta --agree-to-license
    Note that it is very difficult to remove Rosetta 2 once it is installed.
  EOS
end
```

I'm not familiar with the CI but noticed that the runner-images for `macos-14` are both based on `macos-14-arm64` and the ones for `macos-13` and `macos-12` seem to be based in intel (`macos-13` and `macos-12` respectively).
Depending on how `RUBY_PLATFORM` is set here,
```ruby
mentions_rosetta = cask.caveats.include?("requires Rosetta 2")
```
would evaluate to `false` and fail the audit.

**Comment by khipp** [comment]
@bevanjkay Can we reopen this to prevent new PRs until the issue is resolved? The latest version is now `17.4`.

File: Casks/c/cocktail.rb [hunk 1]
```diff
@@ -88,8 +88,8 @@
     end
   end
   on_sonoma :or_newer do
-    version "17.3"
-    sha256 "95de3f2b202ceb2bea74004353d7cc349d88373319488d52fc47ebe160508032"
+    version "17.3.1"
+    sha256 "380f64edc0045d42e5509c0f6d6d1b7ff4c547540ac6f9c6a8a1b2be101b6455"
```
Comment (bevanjkay) [inline Casks/c/cocktail.rb:92]: @khipp In all of these cases we will need to use `:no_check` because the `url` only uses `version.major`.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**